scrapy/squeues.py

```
@@ -88,6 +97,93 @@ def _pickle_serialize(obj):
         raise ValueError(str(e)) from e
 
 
+class _RedisQueue(ABC):
+
+    def __init__(self, path, settings=None):
+        try:
+            import redis  # noqa: F401
+        except ImportError:
+            raise NotConfigured('missing redis library')
+
+        # If called from from_crawler() method, self.crawler is set.
+        # If called from from_settings() method, settings is given.
+        if not settings:
+            settings = self.crawler.settings
+
+        self.client = redis.Redis(
+            host=settings['SCHEDULER_EXTERNAL_QUEUE_REDIS_HOST'],
```

Not completely sure, but better to use `settings.get`, otherwise we can get a `KeyError`.
If we use `settings.get` and it is `None` then we can raise a specific exception with the missing setting and how to solve it

---

Alright, I changed this and raise a `NotConfigured` exception.